src/Rings/MPolyQuo.jl

```diff
@@ -106,28 +106,22 @@ end
   qRing::MPolyQuoRing
 
   function MPolyQuoIdeal(Ox::MPolyQuoRing{T}, si::Singular.sideal) where T <: MPolyRingElem
-   singular_quotient_ring(Ox) == base_ring(si) || error("base rings must match")
+   @req singular_quotient_ring(Ox) == base_ring(si) "base rings must match"
    r = new{T}()
    r.gens  = IdealGens(Ox, si)
    r.qRing = Ox
    r.dim   = -1
    R = base_ring(Ox)
    r.gens.O = [R(g) for g = gens(r.gens.S)]
    B = r.gens
-   if length(B) >= 1
-     if R isa MPolyDecRing
-       if is_graded(R)
-         if !(all(is_homogeneous, B.gens.O))
-            throw(ArgumentError("The generators of the ideal must be homogeneous."))
-         end
-       end
-     end
+   if length(B) >= 1 && is_graded(R)
+     @req all(is_homogeneous, B.gens.O) "The generators of the ideal must be homogeneous"
    end
    return r
   end
 
   function MPolyQuoIdeal(Ox::MPolyQuoRing{T}, I::MPolyIdeal{T}) where T <: MPolyRingElem
-    base_ring(Ox) === base_ring(I) || error("base rings must match")
+    @req base_ring(Ox) === base_ring(I) "base rings must match"
```

Comment:
e.g. here we compare base rings with `===` ...

Reply:
It is the same as `==`.